### PR TITLE
Grt bad boy

### PR DESCRIPTION
**Pull request by seiliki**

### CODE REVIEW

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	}
+	
+	// Update is called once per frame
+	void Update () {
+		if (Input.GetButtonDown("Fire1")){  //randomize bark pitch bark, and ramp up the rattlin'
+            lastFrameRattled = Time.frameCount;
+            barkSound.pitch = 1.2f + GenerateNormalRandom(0f, .05f, -1, 1);
+            barkSound.Play();
+            faceRenderer.sprite = barkingface;
+            this.rattleCount += 1;
```
Comment (kocosephia) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:34]: don't need to "this" if you don't want to

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        float rand1 = Random.Range(0.0f, 1.0f);
+        float rand2 = Random.Range(0.0f, 1.0f);
+        float n = Mathf.Sqrt(-2.0f * Mathf.Log(rand1)) * Mathf.Cos((2.0f * Mathf.PI) * rand2);
+        float generatedNumber = mean + sigma * n;
+        generatedNumber = Mathf.Clamp(generatedNumber, min, max);
+        return generatedNumber;
+    }
+
+    public override void Won() { }
```
Comment (kocosephia) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:71]: why

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        this.faceRenderer = GetComponent<SpriteRenderer>();
+        this.barkSound = GetComponent<AudioSource>();
+
+	}
+	
+	// Update is called once per frame
+	void Update () {
+		if (Input.GetButtonDown("Fire1")){  //randomize bark pitch bark, and ramp up the rattlin'
+            lastFrameRattled = Time.frameCount;
+            barkSound.pitch = 1.2f + GenerateNormalRandom(0f, .05f, -1, 1);
```
Comment (kocosephia) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:31]: this looks like some good wacky shit here.

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        this.transform.Translate(new Vector2(
+            GenerateNormalRandom(0f, rattle/10, -rattle, rattle),
+            GenerateNormalRandom(0f, rattle/10, -rattle, rattle)
+            )
+        );
+    }
+
+    // I KNOW
+    private static float GenerateNormalRandom(float mean, float sigma, float min, float max)
```
Comment (kocosephia) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:61]: Do you not think we all deserve access to your statistical magic?

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs (hunk 3 above)
Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:31]: it sounds better at 1.2

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs (hunk 4 above)
Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:61]: no but it should probably be like a library

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs (hunk 1 above)
Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:34]: why this anything

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs (hunk 2 above)
Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:71]: I thought I would get in trouble if I didn't override my interface

File: WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs (hunk 3 above)
Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:31]: and then it's like a little different

Comment (seiliki) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Prefabs/Minigames/Barkscript.cs:31]: everytime

**Pull request merged**